lambda/custom/helperFunctions.js

```diff
@@ -935,6 +935,47 @@ const customLog = async (data) => {
 	}
 };
 
+const leaveChannel = async (roomId, roomname, type, headers) => {
+	try {
+		let response;
+		if (type === 'c') {
+			response = await axios.post(apiEndpoints.leavechannelurl, {
+				roomId,
+			},
+			{
+				headers,
+			}).then((res) => res.data);
+		} else if (type === 'p') {
+			response = await axios.post(apiEndpoints.leavegroupurl, {
+				roomId,
+			},
+			{
+				headers,
+			}).then((res) => res.data);
+		}
+
+		if (response.success) { return ri('LEAVE_CHANNEL.SUCCESS', { roomname }); }
+
+		return ri('LEAVE_CHANNEL.ERROR');
+
+	} catch (err) {
+		// console.log(err);
+		if (err.response.data.errorType && err.response.data.errorType === 'error-user-not-in-room') {
+			return ri('LEAVE_CHANNEL.USER_NOT_IN_ROOM', { roomname });
+		} else if (err.response.data.errorType && err.response.data.errorType === 'error-you-are-last-owner') {
+			return ri('LEAVE_CHANNEL.ERROR_LAST_OWNER');
+		} else if (err.response.data.errorType && err.response.data.errorType === 'error-room-not-found') {
+			return ri('LEAVE_CHANNEL.ERROR_NOT_FOUND', { roomname });
+		} else if (err.response.data.errorType && err.response.data.errorType === 'error-invalid-room') {
+			return ri('LEAVE_CHANNEL.ERROR_NOT_FOUND', { roomname });
+		} else if (err.response.status === 401) {
+			return ri('AUTH_ERROR');
+		} else {
+			return ri('LEAVE_CHANNEL.ERROR');
```

Comment:
Maybe should add console.log(err) there?

Reply:
Done 👍